### PR TITLE
chore(e2e): retry ttfq test to work around the flake COMPASS-7374

### PR DESCRIPTION
**Pull request by gribnoysup**
Doesn't look like we can easily get to any useful info to actually debug the root cause why selenium (or devtools server?) fails to start in some cases, so we will just retry this test. If the issue resurfaces, then we can invest more time investigating

### CODE REVIEW

File: packages/compass-e2e-tests/helpers/compass.ts [hunk 1]
```diff
@@ -153,6 +153,7 @@ export class Compass {
   }
 
   async recordLogs(): Promise<void> {
+    debug('Setting up renderer log listeners ...');
```
Comment (gribnoysup) [inline packages/compass-e2e-tests/helpers/compass.ts:156]: This was just helpful to understand that it's not trying to get the puppeteer connection that fails, but actually starting the remote webdriver session

File: packages/compass-e2e-tests/tests/time-to-first-query.test.ts [hunk 2]
```diff
@@ -22,6 +22,16 @@ describe('Time to first query', function () {
...
   it('can open compass, connect to a database and run a query on a collection (never seen welcome)', async function () {
+    // Starting the application with the webdriver.io fails on the first run
+    // sometimes due to devtools / selenium server failing to start in time.
+    // While the root cause is unknown, it usually passes just fine on a re-run
+    // or next application start, so we are just retrying the test here to
+    // work around the flake.
+    //
+    // We re-run the whole test to make sure that the timings for the test run
+    // are not skewed by waiting for the application to restart multiple times.
+    this.retries(5);
```
Comment (addaleax) [inline packages/compass-e2e-tests/tests/time-to-first-query.test.ts:33]: Huh, TIL that you can do this :slightly_smiling_face:

**Review by addaleax — APPROVED** [review]

File: packages/compass-e2e-tests/helpers/compass.ts [hunk 3]
```diff
@@ -616,7 +620,67 @@ async function startCompass(opts: StartCompassOptions = {}): Promise<Compass> {
...
+    const { default: psList }: typeof import('ps-list') = await eval(
+      `import('ps-list')`
+    );
+    const processList = await psList();
+
+    const filteredProcesses = processList.filter((p) => {
+      return (
+        p.ppid === process.pid &&
+        (p.cmd?.startsWith(binary) ||
+          /(MongoDB Compass|Electron|electron)/.test(p.name))
```
Comment (gribnoysup) [inline packages/compass-e2e-tests/helpers/compass.ts:647]: Maybe an overkill, but cmd is marked as optional in package types, so I'm a bit afraid to only rely on it for the filtering

**Comment by gribnoysup** [comment]
@addaleax do you mind taking another look at the process clean up code when you have a moment? This seems to work consistently now after a bit of adjusting to cover different cases of how the hanging app shows up in the process list

File: packages/compass-e2e-tests/helpers/compass.ts [hunk 4]
```diff
@@ -616,7 +620,67 @@ async function startCompass(opts: StartCompassOptions = {}): Promise<Compass> {
...
+        : `Found following applications running: ${filteredProcesses
+            .map((p) => {
+              return p.name;
+            })
+            .join(', ')}`
+    );
+
+    filteredProcesses.forEach((p) => {
+      try {
+        debug(`Killing proces ${p.name} with PID ${p.pid}`);
```
Comment (addaleax) [inline packages/compass-e2e-tests/helpers/compass.ts:670]:
```suggestion
        debug(`Killing process ${p.name} with PID ${p.pid}`);
```

**Review by addaleax — APPROVED** [review]

**Pull request merged**